BlueprintUI/Sources/Environment/EnvironmentKey.swift

```diff
@@ -26,4 +26,88 @@ public protocol EnvironmentKey {
     /// The default value that will be vended by an `Environment` for this key if no other value
     /// has been set.
     static var defaultValue: Self.Value { get }
+
+
+    /// Compares two environment values without direct conformance of the values.
+    /// - Parameters:
+    ///   - lhs: The left hand side value being compared.
+    ///   - rhs: The right hand side value being compared.
+    ///   - context: The context to evaluate the equivalency.
+    /// - Returns: Whether or not the two values are equivalent in the specified context.
+    static func isEquivalent(lhs: Value, rhs: Value, in context: CrossLayoutCacheableContext) -> Bool
```

Comment:
I still feel like it's a mistake to not have a default implementation here, but it's not a blocker.